### DISCUSSION TITLE
How do I enable all stable feature flags during upgrades on Kubernetes?

### DISCUSSION THREAD

**Discussion by manueljishi** [rabbitmq/rabbitmq-server#11887] [Ideas] [answered]
### Is your feature request related to a problem? Please describe.

When upgrading rabbitmq-server, in kubernetes my case I stomped with the issue that all feature flags must be enabled before the upgrade. Come to find that apparently the only way to enable all stable feature flags at once is by ussing the `rabbitmqctl enable_feature_flag all` has to be run on every version

### Describe the solution you'd like

I would like to have an option to enable all feature flags at the config level so the configuration will stay the same even if I delete my pods or deploy a new version.
I'm aware of the RABBITMQ_FEATURE_FLAGS env variable, however in the docs it's never mentioned that you can add "all" to it and all feature flags will be enabled, it seems like it only accepts a list.
Maybe I am missing something? 
Thank you very much

**Comment by michaelklishin** [comment, marked as answer]
@manueljishi `RABBITMQ_FEATURE_FLAGS` is not documented because it's not supposed to be used by the users.

Feature flags are not your regular `rabbitmq.conf` settings. They depend on cluster membership, as in, they are a part of cluster state and in some scenarios you won't be able to enable a feature flag because not all nodes currently in the cluster support it.

Our [Kubernetes Operator](https://www.rabbitmq.com/kubernetes/operator/operator-overview) both provides a [`kubectl` extension that allows you to enable the flags](https://www.rabbitmq.com/kubernetes/operator/operator-overview).

The same K8S Operator [enables all feature flags at the end of a rolling deployment](https://www.rabbitmq.com/kubernetes/operator/operator-overview#limitations).

If you DIY your Kubernetes installation, it's up to you to take care of those things the same way the Operator does.
